src/scalar/timeintegrate_scalar_mod.F90

```diff
@@ -50,7 +50,7 @@ SUBROUTINE timeintegrate_scalar(itstep, ittot, timeph, dt, irk, rkscheme)
             CALL get_field(dt_f, "D"//TRIM(scalar(l)%name))
             CALL get_field(told, TRIM(scalar(l)%name)//"_OLD")
 
-            ! Copy to "T_OLD"
+            ! Copy to "T_OLD"  (TO DO: Is this step needed?)
```

Comment:
Yes, the Bossinesq term needs the scalar value from the previous timestep. Since the scalar is integrated before the flow we need to store the old one. Then, when the flow comes next all the terms used to compute the RHS for the flow is from the same time level.

Reply:
Perfect. Thanks for clarification.